internal/handshake/crypto_setup_client.go

```diff
@@ -198,15 +198,15 @@ func (h *cryptoSetupClient) handleREJMessage(cryptoData map[Tag][]byte) error {
 		err = h.certManager.Verify(h.hostname)
 		if err != nil {
 			utils.Infof("Certificate validation failed: %s", err.Error())
-			return qerr.ProofInvalid
+			return qerr.Error(qerr.ProofInvalid, err.Error())
```

Comment:
@lucas-clemente: Do you see any risk with sending the error message in the CONNECTION_CLOSE?

Reply:
No, but given that most TLS implementations are very silent about errors I'd rather not include it.